corehq/apps/app_manager/static/app_manager/js/manage_releases.js

```diff
@@ -0,0 +1,101 @@
+            self.errorMessage = ko.observable();
+            self.domId = "restriction_" + self.id;
+            self.ajaxInProgress = ko.observable(false);
+            self.actionText = ko.computed(function () {
+                return (self.active() ? gettext("Remove") : gettext("Add"));
+            });
+            self.toggleStatus = function () {
+                self.active(!self.active());
+            };
+            self.error = ko.observable();
+            self.requestUrl = function () {
+                if (self.active()) {
+                    return initialPageData.reverse('deactivate_release_restriction', self.id);
+                }
+                return initialPageData.reverse('activate_release_restriction', self.id);
+            };
+            self.toggleRestriction = function () {
+                self.ajaxInProgress(true);
+                var oldStatus = self.active();
+                $.ajax({
+                    method: 'POST',
+                    url: self.requestUrl(),
+                    success: function (data) {
+                        if (data.success) {
+                            self.toggleStatus();
+                            self.activatedOn(data.activated_on);
+                            self.deactivatedOn(data.deactivated_on);
+                            self.error(false);
+                        } else {
+                            self.active(oldStatus);
+                            self.errorMessage(data.message);
+                        }
+                    },
+                    error: function () {
+                        self.active(oldStatus);
+                    },
+                    complete: function () {
+                        self.ajaxInProgress(false);
+                        if (self.active() === oldStatus) {
+                            self.error(true);
+                        }
+                    },
+                });
+            };
+            return self;
+        };
+
+        function manageReleasesViewModel(enabledAppReleases) {
+            var self = {};
+            self.enabledAppReleases = ko.observableArray(enabledAppReleases);
+            return self;
+        }
+        var enabledAppReleases = _.map(initialPageData.get('enabled_app_releases'), function (appRelease) {
+            return enabledAppRelease(appRelease);
+        });
+        var viewModel = manageReleasesViewModel(enabledAppReleases);
+        if (enabledAppReleases.length) {
+            $('#managed-releases').koApplyBindings(viewModel);
+        }
+        function manageReleaseSearchViewModel() {
+            var self = {};
+            self.search = function () {
+                var appId = $("#app-id-search-select").val();
+                var locationId = $("#location-search-select").val();
+                var version = $("#version-input").val();
+                window.location.search = ("location_id=" + locationId + "&app_id=" + appId + "&version=" +
```

Comment:
This forces the page to reload with these GET parameters, right? Any reason not to set this up as a form instead?

Reply:
you mean add a new form?
i would like to avoid doing this JS hack if possible while keeping a single form.

Reply:
I do mean a new form, curious why you'd rather not add one? I think it could be an HTML form - just doing things like wrapping the HTML inputs in a `<form data-bind="submit: search">` and then using [serialize](https://api.jquery.com/serialize/) - not a full Django-based form.

Reply:
i like how this single form is doing both. Don't mind adding one if that adds a value/clarity

Reply:
Eh, I don't feel strongly about it.